src/main/java/za/co/neslotech/spring/trainspringbootinsight/handler/RestExceptionHandler.java

```diff
@@ -14,13 +16,19 @@
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 import za.co.neslotech.spring.trainspringbootinsight.exception.InvalidAuthorizationRequest;
 
+import java.nio.file.AccessDeniedException;
+
 @ControllerAdvice
 public class RestExceptionHandler extends ResponseEntityExceptionHandler {
 
     @ExceptionHandler(EntityNotFoundException.class)
     protected ResponseEntity<Object> handleEntityNotFound(final EntityNotFoundException exception,
                                                           final WebRequest request) {
-        return handleExceptionInternal(exception, exception.getMessage(), new HttpHeaders(),
+
+        final var headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+
+        return handleExceptionInternal(exception, String.format("{ \"message\": \"%s\"}", exception.getMessage()), headers,
```

Comment:
So an alternative to creating a formatted string response json object, you can create a generic API error object that can consume some information like the error message, the reason for the error, the response type, etc. You can then just return a repsponse entity of the error type with all the information and it should then be mapped to json.